pkg/middleware/stored_session.go

```diff
@@ -252,3 +262,40 @@ func (s *storedSessionLoader) validateSession(ctx context.Context, session *sess
 
 	return nil
 }
+
+// forceRefreshSession attempts to load a session as identified by the request cookies.
+// If no session is found, the request will be passed to the next handler.
+// If a session was loaded by a previous handler, it will not be refreshed.
+func (s *storedSessionLoader) forceRefreshSession(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		scope := middlewareapi.GetRequestScope(req)
+
+		// If refreshPeriod isn't defined then forward request to the next handler
+		if s.refreshPeriod <= time.Duration(0) {
+			next.ServeHTTP(rw, req)
+			return
+		}
+
+		session, err := s.store.Load(req)
+		if err == nil && session != nil {
+			err = s.refreshSessionIfNeeded(rw, req, session, true)
+		}
+
+		if err != nil {
+			session = nil
+			if !errors.Is(err, http.ErrNoCookie) {
+				// In the case when there was an error loading the session,
+				// we should clear the session
+				logger.Errorf("Error loading cookied session: %v, removing session", err)
+				err = s.store.Clear(rw, req)
+				if err != nil {
+					logger.Errorf("Error removing session: %v", err)
+				}
+			}
+		}
+
+		// Add the session to the scope if it was found
+		scope.Session = session
+		next.ServeHTTP(rw, req)
+	})
+}
```

Comment:
```suggestion
func (s *storedSessionLoader) forceRefreshSession(next http.Handler) http.Handler {
	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
		scope := middlewareapi.GetRequestScope(req)

		// If refreshPeriod isn't defined then forward request to the next handler
		if s.refreshPeriod <= time.Duration(0) {
			next.ServeHTTP(rw, req)
			return
		}

		session, err := s.store.Load(req)
		if err != nil {
			handleRefreshError(next)
			return
		}
			
		if err = s.refreshSessionIfNeeded(rw, req, session, true); err != nil {
			handleRefreshError(next)
			return
		}

		scope.Session = session
		next.ServeHTTP(rw, req)
	})
}

func (s *storedSessionLoader) handleRefreshError(next http.Handler) {
	if errors.Is(err, http.ErrNoCookie) {
		// what happens if the error is of type http.ErrNoCookie?
	}

	// In the case when there was an error loading the session,
	// we should clear the session
	logger.Errorf("Error loading cookied session: %v, removing session", err)
	err = s.store.Clear(rw, req)
	if err != nil {
		logger.Errorf("Error removing session: %v", err)
	}

	scope.Session = nil
	next.ServeHTTP(rw, req)

```

Reply:
how about this? (I haven't tested it)

Reply:
I'm not against it, but I think you are complicating the code and the code would require updating to work.  The original logic closely follows the same flow that the loadSession function uses.  I'm not sure why this style question is causing a holdup.

Reply:
@tuunit / @JoelSpeed  the requested changes have been made.  Please review